editor/src/clj/editor/pipeline.clj

```diff
@@ -109,10 +109,23 @@
 
 (defn- make-dep-resources
   [deps build-targets-by-content-hash]
+  ;; Create a map that resolves an original BuildResource into a fused
+  ;; BuildResource. Build target fusion is based on the content-hash values of
+  ;; the build targets. In order to fuse build targets from both editable and
+  ;; non-editable BuildResources, we make sure to add their counterpart to the
+  ;; resulting map alongside the original BuildResource.
   (into {}
-        (map (fn [{:keys [content-hash resource] :as _build-target}]
-               (assert (bt/content-hash? content-hash))
-               [resource (:resource (get build-targets-by-content-hash content-hash))]))
+        (mapcat
+          (fn [{:keys [content-hash] :as build-target}]
+            (assert (bt/content-hash? content-hash))
+            (let [original-build-resource (:resource build-target)
+                  counterpart-build-resource (workspace/counterpart-build-resource original-build-resource)
+                  fused-build-target (get build-targets-by-content-hash content-hash)
+                  fused-build-resource (:resource fused-build-target)]
+              (cond-> [(pair original-build-resource fused-build-resource)]
+
+                      counterpart-build-resource
+                      (conj (pair counterpart-build-resource fused-build-resource))))))
```

Comment:
This was the source of the exception. The build targets from both embedded resources will have the same `content-hash`, but in `dep-resources`, only the `:resource` of whichever of them got put in `build-targets-by-content-hash` last would resolve to the matching build target.